File: src/csharp/Intel/Iced.UnitTests/Intel/InstructionInfoTests/InstructionInfoTestCase.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if INSTR_INFO
using System;
using System.Collections.Generic;
using Iced.Intel;

namespace Iced.UnitTests.Intel.InstructionInfoTests {
	public sealed class InstructionInfoTestCase {
		public InstructionInfoTestCase() => Static.Assert(IcedConstants.MaxOpCount == 5 ? 0 : -1);
		public ulong IP = 0;
		public EncodingKind Encoding = EncodingKind.Legacy;
		public CpuidFeature[] CpuidFeatures = Array.Empty<CpuidFeature>();
		public RflagsBits RflagsRead = RflagsBits.None;
		public RflagsBits RflagsUndefined = RflagsBits.None;
		public RflagsBits RflagsWritten = RflagsBits.None;
		public RflagsBits RflagsCleared = RflagsBits.None;
		public RflagsBits RflagsSet = RflagsBits.None;
		public int StackPointerIncrement = 0;
		public bool IsPrivileged = false;
		public bool IsStackInstruction = false;
		public bool IsSaveRestoreInstruction = false;
		public bool IsSpecial = false;
		public readonly List<UsedRegister> UsedRegisters = new List<UsedRegister>();
		public readonly List<UsedMemory> UsedMemory = new List<UsedMemory>();
		public FlowControl FlowControl = FlowControl.Next;
		public OpAccess Op0Access = OpAccess.None;
		public OpAccess Op1Access = OpAccess.None;
		public OpAccess Op2Access = OpAccess.None;
		public OpAccess Op3Access = OpAccess.None;
		public OpAccess Op4Access = OpAccess.None;
		public int FpuTopIncrement = 0;
		public bool FpuConditionalTop = false;
		public bool FpuWritesTop = false;
	}
}
#endif

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/info/InstructionInfoTestCase.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.info;

import java.util.ArrayList;

import com.github.icedland.iced.x86.EncodingKind;
import com.github.icedland.iced.x86.FlowControl;
import com.github.icedland.iced.x86.RflagsBits;
import com.github.icedland.iced.x86.internal.IcedConstants;

final class InstructionInfoTestCase {
	InstructionInfoTestCase() {
		int value = 5;
		if (IcedConstants.MAX_OP_COUNT != value)
			throw new UnsupportedOperationException();
	}
	int lineNo;
	String hexBytes;
	int code;
	int options;
	long ip = 0;
	int encoding = EncodingKind.LEGACY;
	int[] cpuidFeatures = new int[0];
	int rflagsRead = RflagsBits.NONE;
	int rflagsUndefined = RflagsBits.NONE;
	int rflagsWritten = RflagsBits.NONE;
	int rflagsCleared = RflagsBits.NONE;
	int rflagsSet = RflagsBits.NONE;
	int stackPointerIncrement = 0;
	boolean isPrivileged = false;
	boolean isStackInstruction = false;
	boolean isSaveRestoreInstruction = false;
	boolean isSpecial = false;
	final ArrayList<UsedRegister> usedRegisters = new ArrayList<UsedRegister>();
	final ArrayList<UsedMemory> usedMemory = new ArrayList<UsedMemory>();
	int flowControl = FlowControl.NEXT;
	int op0Access = OpAccess.NONE;
	int op1Access = OpAccess.NONE;
	int op2Access = OpAccess.NONE;
	int op3Access = OpAccess.NONE;
	int op4Access = OpAccess.NONE;
	int fpuTopIncrement = 0;
	boolean fpuConditionalTop = false;
	boolean fpuWritesTop = false;
}
